### DISCUSSION TITLE
The role of ABC?

### DISCUSSION THREAD

**Discussion by ApeachM** [YosysHQ/yosys#4245] [Q&A]
Hi, YosysHQ.
As you know, the licenses are different each other between yosys and abc.

But I noticed that yosys uses abc when I build the yosys project.
I want to know if I do know use abc. Will it not synthesis the logic? 
Is abc for logic synthesis, or it is only for timing driven logic optimization and technology mapping?

Thank you, very much.

**Comment by nakengelhardt** [comment]
abc is used for technology mapping, and it also does some gate-level optimizations. You can synthesize to internal gate-level representation without abc by using `synth -noabc`, but it will be somewhat lower quality results. There are also some alternatives for lut mapping, but they are much lower quality: `synth -lut -flowmap` has decent results but tends to crash, `synth -lut -noabc` will use one lut per gate which is very inefficient.
